frontend/server/controllers/BadgesController.php

```diff
@@ -0,0 +1,75 @@
+<?php
+require_once 'libs/dao/User_Rank.dao.php';
+
+/**
+ * BadgesController
+ *
+ * @author carlosabcs
+ */
+class BadgesController extends Controller {
+    const OMEGAUP_BADGES_ROOT = OMEGAUP_ROOT . '/badges';
+
+    /**
+     * Returns a list of existing badges
+     *
+     * @param Request $r
+     * @return array
+     * @throws InvalidDatabaseOperationException
+     */
+    public static function apiList(Request $r) {
+        try {
+            self::authenticateRequest($r);
+        } catch (UnauthorizedException $e) {
+            // Do nothing, we allow unauthenticated users to use this API
+        }
+        $aliases = array_diff(scandir(static::OMEGAUP_BADGES_ROOT), ['..', '.', 'default_icon.svg']);
+        $results = [];
+        foreach ($aliases as $alias) {
+            if (!is_dir(static::OMEGAUP_BADGES_ROOT . "/${alias}")) {
+                continue;
+            }
+            $results[] = $alias;
+        }
+        return $results;
+    }
+
+    /**
+     * Returns a list of badges owned by current user
+     *
+     * @param Request $r
+     * @return array
+     * @throws InvalidDatabaseOperationException
+     */
+    public static function apiMyList(Request $r) {
+        self::authenticateRequest($r);
+        return UsersBadgesDAO::getUserOwnedBadges($r['current_user_id']);
```

Comment:
nit:

```suggestion
        return [
            'status' => 'ok',
            'badges' => UsersBadgesDAO::getUserOwnedBadges($r['current_user_id']),
        ];
```

igual abajo. las únicas funciones que pueden regresar un arreglo normal son `apiList`.